hack/reduce_nat_gateway_cost/replace_nat_with_nat_instance.py

```
@@ -801,91 +678,103 @@ def create_nat_instance(ec2_client, vpc_id, nat_gateway_id, public_subnet: Dict,
+                )
+                nat_instance = instance["Instances"][0]
+                break
+            except botocore.exceptions.ClientError as e:
+                if e.response["Error"]["Code"] == "Unsupported":
+                    logger.info(f'{NAT_INSTANCES_INFO[nat_instance_idx].instance_type} is not supported in this region.')
+                    nat_instance_idx += 1
+                else:
+                    raise
+    except Exception as e:
+        # If instance creation fails, clean up the security group we created
+        logger.error(f'Failed to create NAT instance: {e}. Cleaning up security group {nat_sg_id}')
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)